dali/kernels/slice/slice_cpu.h

```diff
@@ -275,9 +278,94 @@ void SliceKernel(OutputType *output,
+void SliceKernel(ExecutionEngine &exec_engine,
+                 OutputType *output,
+                 const InputType *input,
+                 const TensorShape<Dims> &in_strides,
+                 const TensorShape<Dims> &out_strides,
+                 const TensorShape<Dims> &anchor,
+                 const TensorShape<Dims> &in_shape,
+                 const TensorShape<Dims> &out_shape,
+                 const OutputType *fill_values,
+                 int channel_dim = -1,  // negative if no channel dim or already processed
+                 int min_blk_sz = 16000,
+                 int req_nblocks = -1) {
+  // Parallelize
+  if (req_nblocks < 0)
+    req_nblocks = exec_engine.NumThreads() * 8;
+
+  int nblocks = 1;
+  std::array<int, Dims> split_factor;
+  if (req_nblocks > 1) {
+    split_factor.fill(1);
+    // Either ``req_nblocks`` blocks or fewer if remaining block sizes < min_blk_sz
+    split_shape(split_factor, out_shape, req_nblocks, min_blk_sz);
+    nblocks = volume(split_factor);
+  }
+
+  if (nblocks == 1) {
+    SliceKernel(output, input, in_strides, out_strides, anchor, in_shape, out_shape,
+                fill_values, channel_dim);
+    return;
+  }
+
+  int last_split_dim = LastSplitDim(split_factor);
+  TensorShape<Dims> start;
+  const auto& end = out_shape;
+
+  ScheduleBlocks(
+    start, end, split_factor, 0, last_split_dim,
+    [&](const TensorShape<Dims> &blk_start, const TensorShape<Dims> &blk_end) {
+      auto output_ptr = output;
+      TensorShape<Dims> blk_anchor;
+      TensorShape<Dims> blk_shape;
+      for (int d = 0; d < Dims; d++) {
+        output_ptr += blk_start[d] * out_strides[d];
+        blk_shape[d] = blk_end[d] - blk_start[d];
+        blk_anchor[d] = anchor[d] + blk_start[d];
+      }
+      exec_engine.AddWork([=](int) {
+        SliceKernel(output_ptr, input, in_strides, out_strides, blk_anchor,
+                    in_shape, blk_shape, fill_values, channel_dim);
+      }, volume(blk_shape), false);
+    });
+  exec_engine.RunAll();
+}
+
+/**
+ * @brief Specialization for SequentialExecutionEngine.
+ *        The slice is processed without partitioning.
+ */
+template <typename OutputType, typename InputType, int Dims>
+void SliceKernel(SequentialExecutionEngine &exec_engine,
+                 OutputType *output,
+                 const InputType *input,
+                 const TensorShape<Dims> &in_strides,
+                 const TensorShape<Dims> &out_strides,
+                 const TensorShape<Dims> &anchor,
+                 const TensorShape<Dims> &in_shape,
+                 const TensorShape<Dims> &out_shape,
+                 const OutputType *fill_values,
+                 int channel_dim = -1,  // negative if no channel dim or already processed
+                 int min_blk_sz = 16000,
+                 int req_nblocks = -1) {
+  (void) exec_engine;
+  (void) min_blk_sz;
+  (void) req_nblocks;
```

Comment:
Maybe better just to comment out the argument names instead of void cast them (just an opinion).

Reply:
done